nemo/collections/common/parts/preprocessing/collections.py

```diff
@@ -702,18 +702,23 @@ def __init__(
         output_type = self.OUTPUT_TYPE
         data, duration_filtered = [], 0.0
         total_duration = 0.0
+        duration_undefined = True
+
         for audio_file, duration, command, offset in zip(audio_files, durations, labels, offsets):
             # Duration filters.
-            if min_duration is not None and duration < min_duration:
+            if duration is not None and min_duration is not None and duration < min_duration:
                 duration_filtered += duration
                 continue
 
-            if max_duration is not None and duration > max_duration:
+            if duration is not None and max_duration is not None and duration > max_duration:
                 duration_filtered += duration
                 continue
 
             data.append(output_type(audio_file, duration, command, offset))
-            total_duration += duration
+
+            if duration is not None:
+                total_duration += duration
+                duration_undefined = False
```

Comment:
Curious question: what if few samples have duration and some have None in manifest

Reply:
even if the duration of just one sample is specified, the total duration will be calculated and logged

Reply:
But if last sample is None then duration_undefined would be false.

Reply:
Best would be instead of additional `duration_undefined` parameter, just check for `duration=None` and `total_duration` and log accordingly.

Reply:
so, if we have these durations `[1, 1, 2, None]`, shouldn't we report `total_duration=4`? If we check just for one sample, we would report `The durations were not defined`.
Otherwise if some of samples are None, another ones have duration specified, we can log `Total Defined Duration of 4 hours`